frontend/src/metabase/styled-components/containers/GlobalStyles/GlobalStyles.tsx

```diff
@@ -20,6 +21,8 @@ export const GlobalStyles = (): JSX.Element => {
       --color-focus: ${color("focus")};
     }
 
+    ${defaultFontFiles()}
```

Comment:
I'm curious if there are tests for the "metabase on a subpath" scenarios in general - it appears a lot of edge case (e.g. resolving links, font loading) can stem from that. I wonder if that was a source of regression as well.

Reply:
I think this could definitely break it. We need to read the site-url similar to what we're doing here
https://github.com/metabase/metabase/blob/5d436bda63a233a3365bf0635e69e1a3a83408e9/frontend/src/metabase/lib/dom.js#L224-L226

e.g. 
```ts
const siteUrl = useSetting("site-url");
const baseUrl = new URL(siteUrl).pathname.toLowerCase()
```

Reply:
@WiNloSt FYI, I tested this PR on a subpath proxy (e.g. `http://localhost:4000/metabase/dashboard/2-performance-overview`), and the font appears to correctly, from `http://localhost:4000/metabase/app/fonts/Raleway/Raleway-Bold.woff2`.

Tracing the code inside `defaultFontFiles`, the `basePath` here that prefixes the font is `./app/fonts`

```ts
  const localInstanceUrl = baseUrl.endsWith("/") ? baseUrl : `${baseUrl}/`;
  const basePath = `${localInstanceUrl}app/fonts`;
```

![CleanShot 2567-04-19 at 23 11 02@2x](https://github.com/metabase/metabase/assets/4714175/14c553cd-8224-434f-9e6c-b40fd872dea6)

Reply:
I'd prefer to go with the explicit approach that Kelvin proposes - it'll cover edge cases more easily in the long run

Reply:
I'll do one better. I'll just export the `getSitePath` function from `dom.js` for now. I know it uses the old `MetabaseSettings` function but I would rather reuse that function than rewrite the same things